kotlinx-coroutines-core/common/src/flow/operators/Zip.kt

```diff
@@ -174,6 +175,53 @@ internal fun <T, R> Flow<T>.combineLatest(vararg others: Flow<T>, arrayFactory:
     }
 }
 
+/**
+ * Returns a [Flow] whose values are generated with [transform] function every time this flow emits by combining
+ * the value with the latest value emitted by `other`. Emissions are only triggered by this flow, not `other.
+ * It does not call [transform], and the returned flow will not emit, until `other` emits – if it does not emit, the
```

Comment:
_if it does not emit_ -- unclear at this point what "it" refers to.

Reply:
Agreed this wording is confusing. Not sure if this is much better:

Returns a [Flow] whose values are generated by the [transform] function every time this flow emits by combining the value with the latest value emitted by `other`. Emissions are only triggered by this flow, not `other`, and the returned flow will remain active as long as this flow is, and only complete when this flow completes. The `other` flow will be cancelled if this flow completes first. The operator does not call [transform], and the returned flow will not emit, until `other` emits – if `other` does not emit, the returned flow will complete without emitting. However after `other` has emitted it's first value, the returned flow will ignore `other`'s completion and continue to cache the last value it emitted.

Reply:
Minor tweaks:

Returns a [Flow] whose values are generated by the [transform] function every time this flow emits by combining the value with the latest value emitted by other. Emissions are only triggered by this flow, not other, and the returned flow will remain active as long as this flow is, completing only when this flow completes. The other flow will be cancelled if this flow completes first. 

The operator does not call [transform], and the returned flow will not emit, until other emits – if other does not emit, the returned flow will complete without emitting. However after other has emitted its first value, the returned flow will ignore other's completion and continue to cache the last value it emitted.

Reply:
Updated.